<?xml version="1.0"?>
<!--
  Copyright (C) 2018-2021 LEIDOS.

  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->

<!-- 
Example launch file for health monitor node
Loads parameters and configures logging for the node

This launch file is not used in CARMA to start health_monitor 
as it needs to be at the top level in carma_src.launch to be
able to shut down other nodes.
 -->
<launch>

    <!-- Handle global parameters for topics -->
    <remap from="driver_discovery" to="$(optenv CARMA_INTR_NS)/driver_discovery"/>
    <remap from="system_alert" to="/system_alert"/>

    <!-- Handle global parameters for variables-->
    <remap from="~required_drivers" to="/required_drivers"/> 
    <remap from="~lidar_gps_drivers" to="/lidar_gps_drivers"/>
    <remap from="~camera_drivers" to="/camera_drivers"/>
    <remap from="~truck" to="/truck"/>
    <remap from="~car" to="/car"/> 
    
    <!-- Health Monitor Node -->
    <node pkg="health_monitor" type="health_monitor" name="health_monitor" required="true">
        <rosparam command="load" file="$(find health_monitor)/config/params.yaml" />
    </node>
</launch>
